internal/controllers/clusterextension_controller.go

```diff
@@ -112,88 +179,244 @@ func checkForUnexpectedFieldChange(a, b ocv1alpha1.ClusterExtension) bool {
+	}
 
-	if err := r.validateBundle(bundle); err != nil {
-		setInstalledStatusConditionFailed(&ext.Status.Conditions, err.Error(), ext.GetGeneration())
-		setDeprecationStatusesUnknown(&ext.Status.Conditions, "deprecation checks have not been attempted as installation has failed", ext.GetGeneration())
+	bundleFS, err := r.Storage.Load(ctx, ext)
+	if err != nil {
+		apimeta.SetStatusCondition(&ext.Status.Conditions, metav1.Condition{
+			Type:    ocv1alpha1.TypeHasValidBundle,
+			Status:  metav1.ConditionFalse,
+			Reason:  ocv1alpha1.ReasonBundleLoadFailed,
+			Message: err.Error(),
+		})
 		return ctrl.Result{}, err
 	}
 
-	// Assume all bundles are registry+v1 for now, since that's all we support.
-	// If the bundle is not a registry+v1 bundle, the conversion will fail.
-	bundleProvisioner := "core-rukpak-io-registry"
+	chrt, values, err := r.Handler.Handle(ctx, bundleFS, ext)
+	if err != nil {
+		apimeta.SetStatusCondition(&ext.Status.Conditions, metav1.Condition{
+			Type:    ocv1alpha1.TypeInstalled,
+			Status:  metav1.ConditionFalse,
+			Reason:  ocv1alpha1.ReasonInstallationFailed,
+			Message: err.Error(),
+		})
+		return ctrl.Result{}, err
+	}
 
-	// Ensure a BundleDeployment exists with its bundle source from the bundle
-	// image we just looked up in the solution.
-	dep := r.GenerateExpectedBundleDeployment(*ext, bundle.Image, bundleProvisioner)
-	if err := r.ensureBundleDeployment(ctx, dep); err != nil {
-		// originally Reason: ocv1alpha1.ReasonInstallationFailed
+	ac, err := r.ActionClientGetter.ActionClientFor(ctx, ext)
+	if err != nil {
 		ext.Status.InstalledBundle = nil
-		setInstalledStatusConditionFailed(&ext.Status.Conditions, err.Error(), ext.GetGeneration())
-		setDeprecationStatusesUnknown(&ext.Status.Conditions, "deprecation checks have not been attempted as installation has failed", ext.GetGeneration())
+		setInstalledStatusConditionFailed(&ext.Status.Conditions, fmt.Sprintf("%s:%v", ocv1alpha1.ReasonErrorGettingClient, err), ext.Generation)
 		return ctrl.Result{}, err
 	}
 
-	// convert existing unstructured object into bundleDeployment for easier mapping of status.
-	existingTypedBundleDeployment := &rukpakv1alpha2.BundleDeployment{}
-	if err := runtime.DefaultUnstructuredConverter.FromUnstructured(dep.UnstructuredContent(), existingTypedBundleDeployment); err != nil {
-		// originally Reason: ocv1alpha1.ReasonInstallationStatusUnknown
-		ext.Status.InstalledBundle = nil
-		setInstalledStatusConditionUnknown(&ext.Status.Conditions, err.Error(), ext.GetGeneration())
-		setDeprecationStatusesUnknown(&ext.Status.Conditions, "deprecation checks have not been attempted as installation has failed", ext.GetGeneration())
+	post := &postrenderer{
+		labels: map[string]string{
+			labels.OwnerKindKey:     ocv1alpha1.ClusterExtensionKind,
+			labels.OwnerNameKey:     ext.GetName(),
+			labels.BundleNameKey:    bundle.Name,
+			labels.PackageNameKey:   bundle.Package,
+			labels.BundleVersionKey: bundleVersion.String(),
```

Comment:
We should include the catalog in which the bundle was resolved from as well (and support catalog filtering during resolution). There are no guarantees that a bundle is only present/served from a single catalog on the cluster. For example, on OpenShift the Red Hat Operator Catalog and Community Operator Catalog both have some of the same entries (`loki-operator` is one that I found relatively quickly). 

I'm fine with this being done as a follow up if we don't already have support for filtering bundles by catalog name during resolution, but if we do we should ensure we track that information as part of the labels as added by this postrenderer as well IMO.

Reply:
It does not look like there is currently logic to filter resolution to a particular catalog. Not sure it is really in scope for the Helm integration work, but we should probably have an issue tracking this somewhere

Reply:
That's a good point, thanks Bryce! I'll create a follow-up issue for this.

Reply:
#873